modules/gdscript/gdscript_tokenizer.cpp

```diff
@@ -643,6 +663,11 @@ GDScriptTokenizer::Token GDScriptTokenizer::number() {
 	bool has_error = false;
 	bool (*digit_check_func)(char32_t) = is_digit;
 
+	// Sign before hexadecimal or binary.
+	if ((_peek(-1) == '+' || _peek(-1) == '-') && _peek() == '0') {
+		_advance();
+	}
```

Comment:
This is not obvious due to the fact that the first character of the number literal is consumed before the `number()`. I would prefer `number()` to be called when the current character is the first character of the numeric literal, whatever it is (`+`, `-`, `.`, `0` or some other digit).